crabs/tracker/evaluate_tracker.py

```diff
@@ -1,93 +1,112 @@
 import csv
 import logging
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Optional, Tuple
 
 import numpy as np
 
 from crabs.tracker.utils.tracking import extract_bounding_box_info
 
 
 class TrackerEvaluate:
-    def __init__(self, gt_dir: str, tracked_list: list, iou_threshold: float):
-        self.gt_dir = gt_dir
-        self.tracked_list = tracked_list
-        self.iou_threshold = iou_threshold
-
-    def create_gt_list(
+    def __init__(
         self,
-        ground_truth_data: list[Dict[str, Any]],
-        gt_boxes_list: list[np.ndarray],
-    ) -> list[np.ndarray]:
+        gt_dir: str,
+        predicted_boxes_id: list[np.ndarray],
+        iou_threshold: float,
+    ):
         """
-        Creates a list of ground truth bounding boxes organized by frame number.
+        Initialize the TrackerEvaluate class with ground truth directory, tracked list, and IoU threshold.
 
         Parameters
         ----------
-        ground_truth_data : list[Dict[str, Any]]
-            A list containing ground truth bounding box data organized by frame number.
-        gt_boxes_list : list[np.ndarray]
-            A list to store the ground truth bounding boxes for each frame.
+        gt_dir : str
+            Directory path of the ground truth CSV file.
+        tracked_list : List[np.ndarray]
+            A list where each element is a numpy array representing tracked objects in a frame.
+            Each numpy array has shape (N, 5), where N is the number of objects.
+            The columns are [x1, y1, x2, y2, id], where (x1, y1) and (x2, y2)
+            define the bounding box and id is the object ID.
+        iou_threshold : float
+            Intersection over Union (IoU) threshold for evaluating tracking performance.
+        """
+        self.gt_dir = gt_dir
+        self.predicted_boxes_id = predicted_boxes_id
+        self.iou_threshold = iou_threshold
+
+    def get_predicted_data(self) -> Dict[int, Dict[str, Any]]:
+        """
+        Convert predicted bounding box and ID into a dictionary organized by frame number.
 
         Returns
         -------
-        list[np.ndarray]:
-            A list containing ground truth bounding boxes organized by frame number.
+        Dict[int, Dict[str, Any]]:
+            A dictionary where the key is the frame number and the value is another dictionary containing:
+            - 'bbox': A numpy array with shape (N, 4) containing coordinates of the bounding boxes
+            [x, y, x + width, y + height] for every object in the frame.
+            - 'id': A numpy array containing the IDs of the tracked objects.
         """
+        predicted_dict: Dict[int, Dict[str, Any]] = {}
+
+        for frame_idx, frame_data in enumerate(self.predicted_boxes_id):
```

Comment:
```suggestion
        for frame_number, frame_data in enumerate(self.predicted_boxes_id):
```

Sorry, maybe I didn't explain myself very well last time re `frame_idx` and `frame_number`. I didn't mean to rename all `frame_number`s to `frame_idx`, since they have a different meaning.

I call `frame_idx` if it's 0-based (aka an index to a list / array). I use `frame_number` if it's 1-based (or basically if we cannot guarantee that it can be used as an index).

Here we need `frame_number` cause it is extracted from the filename. And we don't know if the frame numbers in the files are 0-based or 1-based, but most importantly we don't know if they start in frame 1. If they don't and the first file is `_0343.png` we cannot convert it to an index by subtracting 1, right?